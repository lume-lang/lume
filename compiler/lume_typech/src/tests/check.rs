mod construct;
mod enums;
mod generic;
mod traits;

use crate::assert_typech_snapshot;

#[test]
fn typech_variable_decl_comp() {
    assert_typech_snapshot!(
        "fn foo() {
        let _: Int32 = 1_i32;
    }"
    );
}

#[test]
fn typech_variable_decl_incomp() {
    assert_typech_snapshot!(
        "fn foo() {
        let _: Int32 = false;
    }"
    );
}

#[test]
fn typech_return_value_inline_comp() {
    assert_typech_snapshot!(
        "fn foo() -> Int32 {
        return 0_i32;
    }"
    );
}

#[test]
fn typech_return_value_inline_incomp() {
    assert_typech_snapshot!(
        "fn foo() -> Int32 {
        return true;
    }"
    );
}

#[test]
fn typech_return_value_condition_if() {
    assert_typech_snapshot!(
        "fn foo() -> Int32 {
        if false {
            return true;
        }
        return 0_i32;
    }"
    );
}

#[test]
fn typech_return_value_condition_inf_loop() {
    assert_typech_snapshot!(
        "fn foo() -> Int32 {
        loop {
            return true;
        }
        return 0_i32;
    }"
    );
}

#[test]
fn typech_return_value_condition_pred_loop() {
    assert_typech_snapshot!(
        "fn foo() -> Int32 {
        while true {
            return true;
        }
        return 0_i32;
    }"
    );
}

#[test]
fn typech_return_value_condition_iter_loop() {
    assert_typech_snapshot!(
        "fn foo() -> Int32 {
        for x in [] {
            return true;
        }
        return 0_i32;
    }"
    );
}

#[test]
fn typech_return_value_call_expr() {
    assert_typech_snapshot!(
        "fn foo() -> Boolean {
            return false;
        }

        fn bar() -> Int32 {
            return foo();
        }"
    );
}

#[test]
fn typech_literal_assignment_int() {
    assert_typech_snapshot!(
        "fn foo() {
            1 = false;
        }"
    );
}

#[test]
fn typech_literal_assignment_bool() {
    assert_typech_snapshot!(
        "fn foo() {
            false = false;
        }"
    );
}

#[test]
fn typech_literal_assignment_str() {
    assert_typech_snapshot!(
        "fn foo() {
            \"foo\" = false;
        }"
    );
}
