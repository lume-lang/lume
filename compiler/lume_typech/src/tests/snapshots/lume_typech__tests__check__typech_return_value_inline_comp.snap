---
source: compiler/lume_typech/src/tests/check.rs
description: "fn foo() -> Int32 {\n        return 0_i32;\n    }"
---

